Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader

```diff
@@ -0,0 +1,360 @@
+// Very fast shader that uses the Unity light system.
+// Compiles down to only performing the operations you're actually using.
+// Uses material property drawers rather than a custom editor for ease of maintenance.
+
+// Textured ambient+diffuse:
+// Stats for Vertex shader:
+//        d3d11: 24 avg math (9..44)
+// Stats for Fragment shader:
+//        d3d11: 2 avg math (1..5), 0 avg texture (0..2)
+
+Shader "HoloToolkit/Fast Configurable"
+{
+    Properties
+    {
+		[Header(Base Texture and Color)]
+		[Indent]
+			[Toggle] _UseVertexColor("Vertex Color Enabled?", Float) = 0
+			[Toggle] _UseMainColor("Main Color Enabled?", Float) = 0
+			_Color("Main Color", Color) = (1,1,1,1)		
```

Comment:
Swapping materials could be bad. It's been known in the past to create mem leaks in unity if improperly implemented.